### PR TITLE
Add Streamlit Chat Migration Guide

### PR DESCRIPTION
**Pull request by philippjfr**
Supersedes https://github.com/holoviz/panel/pull/5642

- [x] Review and merge #5682. The `PanelCallbackHandler` example depends on this.
- [x] Upload assets and replace links

### CODE REVIEW

**Comment by philippjfr** [comment]
Not super convinced that we should add the ChatInput and ChatStatus examples.

**Comment by MarcSkovMadsen** [comment]
Questions

- [ ] Should we provide a `ChatInput` component similar to Streamlits `chat_input`?
- [ ] Should we provide a `Status` component similar to Streamlits `status`?

**Comment by MarcSkovMadsen** [comment]
> Not super convinced that we should add the ChatInput and ChatStatus examples.

I assume its because the components are Long and not for beginners?

But since we dont have those components in Panel hos done guide  users wanting to migrate?

**Comment by ahuang11** [comment]
If we have a ChatInput, I'd like it to be like the GitHub text box which lets users input both text and files by drag/drop.

**Comment by MarcSkovMadsen** [comment]
> Isn't ChatInput equivalent to TextInput at the moment?

TextInput is What we have at the moment. But they are not equivalent.

- the ChatInput combines a TextInput and Button.
- the TextInput is cleared (on html side) when entered or submitted.

The ChatInput is easier to use when submitting multiple messages.

I think we carve out the advanced ChatInput you have implemented in the ChatInterface and Expose that @ahuang11

**Comment by ahuang11** [comment]
Maybe in the future; I wouldn't want to hold up the Panel release any longer. Perhaps track it in a new issue?

**Comment by MarcSkovMadsen** [comment]
I've moved the `ChatInput` and `Status` components to https://github.com/holoviz-topics/panel-chat-examples/pull/69. I will document how to migrate 1) A basic chat example 2) a CallbackHandler example..

**Comment by MarcSkovMadsen** [comment]
## Assets

https://github.com/holoviz/panel/assets/42288570/95a078be-24bb-42a1-af07-ab7c631ac040

**Comment by MarcSkovMadsen** [comment]
This one is *ready for review* @philippjfr . I cannot request it from you as you opened the PR.

File: doc/how_to/streamlit_migration/chat.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+if prompt := st.chat_input():
+    st.chat_message("user").write(prompt)
+    with st.chat_message("assistant"):
+        st_callback = StreamlitCallbackHandler(st.container())
+        response = agent.run(prompt, callbacks=[st_callback])
+        st.write(response)
+```
+
+<video controls="" poster="../../_static/images/streamlit_callback_handler.png">
+    <source src="https://user-images.githubusercontent.com/42288570/276558354-95a078be-24bb-42a1-af07-ab7c631ac040.mp4" type="video/mp4" style="max-height: 400px; max-width: 600px;">
```
Comment (philippjfr) [inline doc/how_to/streamlit_migration/chat.md:134]:
```suggestion
    <source src="https://assets.holoviz.org/panel/how_to/streamlit_migration/streamlit_callback_handler.mp4" type="video/mp4" style="max-height: 400px; max-width: 600px;">
```

File: doc/how_to/streamlit_migration/chat.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+async def callback(contents, user, instance):
+    callback_handler = pn.chat.langchain.PanelCallbackHandler(instance)
+    await agent.arun(contents, callbacks=[callback_handler])
+
+pn.chat.ChatInterface(callback=callback).servable()
+```
+
+<video controls="" poster="../../_static/images/panel_callback_handler.png">
+    <source src="https://user-images.githubusercontent.com/42288570/276561076-7ac4fe04-088e-4dea-8dea-400b6bc7f930.mp4" type="video/mp4" style="max-height: 400px; max-width: 600px;">
```
Comment (philippjfr) [inline doc/how_to/streamlit_migration/chat.md:161]:
```suggestion
    <source src="https://assets.holoviz.org/panel/how_to/streamlit_migration/panel_callback_handler.mp4" type="video/mp4" style="max-height: 400px; max-width: 600px;">
```

**Comment by MarcSkovMadsen** [comment]
Thx

**Pull request merged**